Recast/Source/RecastArea.cpp

```diff
@@ -75,8 +58,8 @@ bool rcErodeWalkableArea(rcContext* ctx, int radius, rcCompactHeightfield& chf)
 						{
 							const int nx = x + rcGetDirOffsetX(dir);
 							const int ny = y + rcGetDirOffsetY(dir);
-							const int nidx = (int)chf.cells[nx+ny*w].index + rcGetCon(s, dir);
-							if (chf.areas[nidx] != RC_NULL_AREA)
+							const int nidx = (int)chf.cells[nx + ny*w].index + rcGetCon(s, dir);
+							if (chf.areas[nidx] != fromAreaID)
```

Comment:
Now that you mention that, I'm not sure how this code does not potentially end up indexing invalid area indices, because RC_NOT_CONNECTED is defined as 0x3f, so I would have expected us to end up with invalid high nidx values. Are you sure that rcGetCon() is actually capable of returning RC_NOT_CONNECTED in this situation?